### PR TITLE
Make Faker an optional dependency.

### LINKED ISSUE #390: Make `faker` an optional dependency.
Hey there!

Along the lines of #292, I was wondering if there is any interest in making `faker` an optional dependency. 

One of my current project setups require me to pin `faker<=2.0.4` as a dev dependency, and this conflicts with it being a hard run-time dep for `dataclasses_avroschema`. I think it'd be neat to use this library so please advise on the best way to proceed. If you can't find the time, thats fine too, as I can try to contribute.

### PR DESCRIPTION
**Pull request by aalekhpatel07**
Closes #390 

This is the first draft of making Faker an optional dependency. 

There are some failing tests (related to datetime serde) that don't really seem to touch my changelist so probably best to defer them to you:

```
FAILED tests/model_generator/test_model_generator.py::test_schema_logical_types - assert "from datacla...cimal('3.14')" == "from datacla...cimal('3.14')"
FAILED tests/model_generator/test_model_generator.py::test_field_order - assert "from datacla...lt', 'money']" == "from datacla...lt', 'money']"
FAILED tests/model_generator/test_model_pydantic_generator.py::test_schema_logical_types - assert "from datacla...cimal('3.14')" == "from datacla...cimal('3.14')"
FAILED tests/schemas/test_roundtrip.py::test_roundtrip[logical_types] - AssertionError: assert {'doc': 'Some...pe': 'record'} == {'doc': 'Some...pe': 'record'}
4 failed, 1648 passed, 3 skipped, 5 xfailed in 2.53s
```

These changes seemed the least invasive to me for a decent implementation but may not necessarily be the best. I'm all ears for suggestions.

### CODE REVIEW

File: dataclasses_avroschema/faker.py [hunk 1]
```diff
@@ -1,3 +1,47 @@
-from faker import Faker
+class FakeStub:
```
Comment (marcosschroh) [inline dataclasses_avroschema/faker.py:1]: I think it is better to move the class and `_FAKER_METHODS_TO_STUB` inside the `except` block. The reason is to make it more clear and second to load this code only when the exception occurs

**Review by marcosschroh — COMMENTED** [review]
- The dependency `faker` is still marked as `optional = false` in the [poetry.lock](https://github.com/aalekhpatel07/dataclasses-avroschema/blob/optional-faker/poetry.lock#L729) file
- Could you `rebase` and/or adjust your commits in order to use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#specification)? We follow the CC practice
- Consider adding the documentation in the `README.md` ([extras section](https://github.com/aalekhpatel07/dataclasses-avroschema/tree/optional-faker#extras)) and in [factories_and_fixtures](https://marcosschroh.github.io/dataclasses-avroschema /factories_and_fixtures/) so end users will know that faker is optional and they need to install it

File: dataclasses_avroschema/faker.py (hunk 1 above)
Comment (aalekhpatel07) [reply dataclasses_avroschema/faker.py]: that's an excellent suggestion. Thank you.

**Comment by marcosschroh** [comment]
Could you rebase from master and solve the conflicts?

**Comment by aalekhpatel07** [comment]
one more run?

**Pull request merged**